nixos/modules/services/continuous-integration/jenkins/credentials.nix

```diff
@@ -0,0 +1,247 @@
+
+with lib;
+
+let
+  jenkinsCfg = config.services.jenkins;
+  cfg = config.services.jenkins.credentials;
+
+  _credentialsType = types.submodule {
+    options = {
+      id = mkOption {
+        type = types.str;
+        example = "my-credentials-id";
+        description = ''
+          The internal unique ID by which the credentials to create will be identified from jobs and other configuration.
+        '';
+      };
+      username = mkOption {
+        type = types.str;
+        example = "pamplemousse";
+        description = ''
+          The username part of the credentials to create.
+        '';
+      };
+      password = mkOption {
+        type = types.str;
+        example = "";
+        description = ''
+          The password part of the credentials to create.
+        '';
+      };
+      description = mkOption {
+        type = types.str;
+        default = "";
+        example = "My new credentials.";
+        description = ''
+          The descriptive text associated with the credentials to create.
+        '';
+      };
+    };
+  };
+in
+{
+  options = {
+    services.jenkins.credentials = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether or not to enable declarative management of Jenkins' credentials.
+
+          Credentials managed through the Jenkins WebUI (or by other means) are left unchanged.
+
+          Note that it really is declarative configuration;
+          if you remove a previously defined credentials, the corresponding credentials will be deleted.
+
+          Please see the Jenkins Credentials plugin documentation for more info:
+          <link xlink:href="https://plugins.jenkins.io/credentials">
+          https://plugins.jenkins.io/credentials</link>
+        '';
+      };
+
+      accessUser = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          User id in Jenkins used to reload config.
+        '';
+      };
+
+      accessToken = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          User token in Jenkins used to reload config.
+          WARNING: This token will be world readable in the Nix store. To keep
+          it secret, use the <option>accessTokenFile</option> option instead.
+        '';
+      };
+
+      accessTokenFile = mkOption {
+        default = "";
+        type = types.str;
+        example = "/run/keys/jenkins-credentials-access-token";
+        description = ''
+          File containing the API token for the <option>accessUser</option>
+          user.
+        '';
+      };
+
+      credentials = mkOption {
+        default = [];
+        type = types.listOf _credentialsType;
+        description = ''
+          The list of credentials to create.
+        '';
+      };
+    };
+  };
+
+  config = mkIf (jenkinsCfg.enable && cfg.enable) {
+    assertions = [
+      { assertion = jenkinsCfg.withCLI;
```

Comment:
Instead of having this assertion that withCLI must be true, you can use an absolute path to the `jenkins-cli` program: `${pkgs.jenkins}/bin/jenkins-cli`.